src/qibolab/compilers/compiler.py

```diff
@@ -148,8 +149,13 @@ def compile(self, circuit: Circuit, platform: Platform):
         measurement_map = {}
         channel_clock = defaultdict(int)
 
+        def find_max(channels: list[Channel]):
+            return max(channel_clock[ch.name] for ch in channels)
+
         def qubit_clock(el: QubitId):
-            return max(channel_clock[ch.name] for ch in platform.elements[el].channels)
+            if el in platform.qubits:
+                return find_max(platform.qubits[el].channels)
+            return find_max(platform.couplers[el].channels)
```

Comment:
The usual (irrelevant) optimization:
```py
def qubit_clock(el: QubitId):
    elements = platform.qubits if el in platform.qubits else platform.couplers
    return max(channel_clock[ch.name] for ch in elements.channels)
```

(However, I appreciated that you split the even small function to avoid duplication - it's still little relevant, but I liked it)

Reply:
8be2dbc4639cd45a964049d3e32ce199d8b4696a